src/smolagents/agent_types.py

```
@@ -93,13 +92,16 @@ def __init__(self, value):
             self._raw = PIL.Image.open(BytesIO(value))
         elif isinstance(value, (str, pathlib.Path)):
             self._path = value
-        elif is_torch_available():
-            import torch
+        else:
+            try:
+                import torch
 
-            if isinstance(value, torch.Tensor):
-                self._tensor = value
-            if isinstance(value, np.ndarray):
-                self._tensor = torch.from_numpy(value)
+                if isinstance(value, torch.Tensor):
+                    self._tensor = value
+                if isinstance(value, np.ndarray):
+                    self._tensor = torch.from_numpy(value)
+            except ModuleNotFoundError:
+                pass
```

Won't it create silent errors in some cases to just `pass` here? For instance if I try to initialize an `AgentImage` from a tensor although `torch` is not installed, it will just not do anything, then fail later whenever I'm trying to use this image?

---

Please note that I have preserved the same behavior before and after this PR:
- if `torch` is not available, the condition you mention does nothing (before or after this PR)
- there is a final condition that raises `TypeError(f"Unsupported type for {self.__class__.__name__}: {type(value)}")`

Before: the flow does not enter the elif condition
```python
elif is_torch_available():
    import torch
    if isinstance(value, torch.Tensor):
        self._tensor = value
```
After: `ModuleNotFoundError` is raised and caught, and the flow continues
```python
try:
    import torch
    if isinstance(value, torch.Tensor):
        self._tensor = value
except ModuleNotFoundError:
    pass
```

Note the final condition, before and after this PR:
```python
if self._path is None and self._raw is None and self._tensor is None:
    raise TypeError(f"Unsupported type for {self.__class__.__name__}: {type(value)}")
```